poll/poll.py

```diff
@@ -662,6 +664,12 @@ def get_results(self, data, suffix=''):
         if self.private_results and not self.can_view_private_results():
             detail, total = {}, None
         else:
+        # Get tally from the db directly
+            tally = self.get_tally()
+            if not self.runtime._services['user'].get_current_user().opt_attrs['edx-platform.is_authenticated'] and tally:
```

Comment:
Code readability could be improved by converting first condition into a function.

```suggestion
            if self.is_anonymous_user() and tally:
```

Reply:
Done!